### PR TITLE
GODRIVER-1844 add PoolReadyEvent and ConnectionCheckOutStartedEvent t…

### PR DESCRIPTION
**Pull request by iwysiu**
…o unified test runner

### CODE REVIEW

**Review by divjotarora — COMMENTED** [review]
IIUC, we don't actually publish these events, so this change will allow these event types to be present in `observeEvents` and `storeEventsASEntities` arrays, but they'll never be captured during execution. Are there planned maintenace tests that will fail in this case?

**Comment by iwysiu** [comment]
Yeah, the planned maintenance tests expect these to exist. Though they don't do anything with the events, so I'm not worried about capturing them until the changes that use these events actually get written

File: mongo/integration/unified/event.go [hunk 1]
```diff
@@ -61,6 +67,7 @@ func monitoringEventTypeFromString(eventStr string) (monitoringEventType, bool)
 }
 
 func monitoringEventTypeFromPoolEvent(evt *event.PoolEvent) monitoringEventType {
+	// TODO GODRIVER-1827: add storePoolReady and storeConnectionCheckOutStarted events
```
Comment (benjirewis) [inline mongo/integration/unified/event.go:70]: Why not add these now? Will there be more work than adding new enum variants?

Comment (iwysiu) [reply mongo/integration/unified/event.go:70]: I can, but I feel like it's misleading to add them to the public events package enum before they actually do anything

**Review by divjotarora — APPROVED** [review]

File: mongo/integration/unified/event.go [hunk 2]
```diff
@@ -14,19 +14,25 @@ import (
 
 type monitoringEventType string
 
+// monitioringEventTypes poolReadyEvent and connectionCheckOutStartedEvent are included
```
Comment (benjirewis) [inline mongo/integration/unified/event.go:17]:
```suggestion
// monitoringEventTypes poolReadyEvent and connectionCheckOutStartedEvent are included
```

**Review by benjirewis — APPROVED** [review]
LGTM with longer comment; small typo.

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**